### ISSUE TITLE
Format strings contain newlines

### ISSUE THREAD

**Issue by pitdicker** [cecton/pure-rust-locales#6]
Example from the POSIX locale:
```
% Appropriate date and time representation (%c)
%       "%a %b %e %H:%M:%S %Y"
d_t_fmt "<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U0065>/
<U0020><U0025><U0048><U003A><U0025><U004D>/
<U003A><U0025><U0053><U0020><U0025><U0059>"
```

is converted into:
```rust
        /// `"%a %b %e\n %H:%M\n:%S %Y"`
        pub const D_T_FMT: &str = "%a %b %e\n %H:%M\n:%S %Y";
```

The newlines should not be part of the format string.

**Comment by pitdicker** [comment]
I have never used nom before, and have trouble figuring out what to change to fix this.

**Comment by cecton** [comment]
I will look into it no worry

**Comment by pitdicker** [comment]
Thank you!

**Comment by cecton** [comment]
@pitdicker I think the trailing `/` should also be discarded, right?

**Comment by pitdicker** [comment]
Yes

**Comment by cecton** [comment]
In the end it was a very small fix. Do you mind adding it to your PR?

```
diff --git a/generate-api/src/parser.rs b/generate-api/src/parser.rs
index ec02874..1632ce8 100644
--- a/generate-api/src/parser.rs
+++ b/generate-api/src/parser.rs
@@ -104,6 +104,7 @@ fn parse_str<'a, E: ParseError<&'a str>>(
         map_parser(
             alt((
                 take_while1(|c| c != escape_char && c != '"'),
+                map(preceded(char(escape_char), char('\n')), |_| ""),
                 preceded(char(escape_char), take(1_usize)),
             )),
             unescape_unicode,
```

**Comment by cecton** [comment]
tbh I was rusty, it took me some time to figure out

also had issues with the build for some reason but somehow the issue disappeared I don't know

**Comment by cecton** [comment]
Fixed in #5
